File: programmers/2_level/49993.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string skill, vector<string> skill_trees) {
    int answer = 0;
    
    for(int i=0; i < skill_trees.size(); i++){
        string tmp = "";
        for(int j = 0; j < skill_trees[i].length(); j++){
            for(char k : skill){
                if(k == skill_trees[i][j]){
                    tmp += k;
                    break;
                }
            }
        }
        
        skill_trees[i] = tmp;
    }
    
    for(int i=0; i < skill_trees.size(); i++){
        if(skill.substr(0,skill_trees[i].length()) == skill_trees[i]) answer++;
    }
    
    return answer;
}

File: programmers/2_level/49993.py
def solution(skill, skill_trees):
    answer = 0
    
    for tree in skill_trees:
        skill_subset = ''.join([ch for ch in tree if ch in skill])
        
        if skill.startswith(skill_subset):
            answer+=1
            
    return answer

"""
from collections import deque


def solution(skill, skill_trees):
    answer = 0

    for seq in skill_trees:
        dq = deque(skill)

        for ch in seq:
            if ch in skill and ch != dq.popleft():
                break
        else:
            answer += 1

    return answer
"""
